File: SCALA/Cap05/c05ex10.scala
// c05ex10.scala

import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    var RESP: Char = 'S'
    var N, R: Int = 0
    var saida: Boolean = true

    do {
      print("Entre um valor numerico: ")
      N = readInt()

      R = N * 3

      println("Resultado = " + R + "\n")

      print("Deseja continuar? (S/N): ")
      RESP = Character.toUpperCase(readChar())

      println()
      if (Character.toUpperCase(RESP) != 'S') {saida = false}
    } while (saida == true)

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}

File: DART/Cap05/c05ex10.dart
// c05ex10.dart

import 'dart:io';

void main() {
  String RESP;
  int N;
  int R;

  do {
    stdout.write("Entre um valor numerico: ");
    N = int.tryParse(stdin.readLineSync() ?? '') ?? 0;

    R = N * 3;

    print("Resultado = $R\n");

    stdout.write("Deseja continuar? (S/N): ");
    RESP = stdin.readLineSync()?.toUpperCase() ?? 'N';

    print("");
    if (RESP != 'S') break;
  } while (true);

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
